unittests/AST/StructuralEquivalenceTest.cpp

```diff
@@ -772,6 +772,20 @@ TEST_F(StructuralEquivalenceRecordTest, RecordsWithDifferentBody) {
   EXPECT_FALSE(testStructuralMatch(t));
 }
 
+TEST_F(StructuralEquivalenceRecordTest, SameFriendMultipleTimes) {
+  auto t = makeNamedDecls("struct foo{ friend class X; };",
+                          "struct foo{ friend class X; friend class X; };",
+                          Lang_CXX);
+  EXPECT_FALSE(testStructuralMatch(t));
```

Comment:
I think this is correct to indicate mismatch. If we think about structural eq as the tool to discover ODR errors than certainly it is correct. If the two entities are defined in different TUs, then they must have the exact same sequence of tokens and the meaning of those tokens should be the same.

Reply:
The test was added to show (and verify) that the current struct eq implementation works this way (even if this may be not correct for all use cases).